packages/dev/core/src/Maths/math.vector.ts

```diff
@@ -1107,18 +1107,54 @@ export class Vector3 {
      * Example Playground https://playground.babylonjs.com/#R1F8YU#57
      * @param scale defines the multiplier factor
      * @param result defines the Vector3 object where to store the result
-     * @returns the current Vector3
+     * @returns the result
      */
     public scaleToRef(scale: number, result: Vector3): Vector3 {
         return result.copyFromFloats(this._x * scale, this._y * scale, this._z * scale);
     }
 
+    /**
+     * Creates a vector normal (perpendicular) to the current Vector3
+     * returns the vector normal
+     */
+    public getNormal(): Vector3 {
+        const ref: Vector3 = MathTmp.Vector3[0];
+        return this.getNormalToRef(ref);
+    }
+
+    /**
+     * Creates a vector normal (perpendicular) to the current Vector3 and stores the result in the given vector
+     * @param result defines the Vector3 object where to store the resultant normal
+     * returns the result
+     */
+    public getNormalToRef(result: DeepImmutable<Vector3>): Vector3 {

```

Comment:
is there not an infinity of possibility ???

Reply:
There is any of which will allow a rotation around it as axis of the cuurent vector about Math.PI radians which is what I need. This function produces one such normal.

Its use is in `RotationQuaternionFromOntoToRef` about line 4198.

Perhaps as it is only designed for use within this method that if you think it better I could remove the `getNormal` method all together and make the `getNormalToRef` method a `@hidden` one and rename it as `_getNormalToRef` to show its not for the user.

Also in this case would there be any benefit to changing to to a static method?

Reply:
Yup I would rather keep it private as it is not clear what other use it would have.

There is no difference with static.

Reply:
Will make changes tomorrow. Let you know when done.

Reply:
Done